### ISSUE TITLE
False positive mismatched types for returned value in function with`todo!()`

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#9566]
## Environment

* **IntelliJ Rust plugin version:** 0.4.182.4945-223-nightly
* **Rust toolchain version:** 1.66.0-nightly (c97d02cdb 2022-10-05) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.3 EAP (CL-223.6646.99)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description
<img width="524" alt="image" src="https://user-images.githubusercontent.com/49211026/196395874-ac8e3979-50aa-4cb0-a197-823ee9314748.png">

## Steps to reproduce
```rust
fn foooo() -> u8 {
    todo!();
    let t = 8;
    println!("end of function");
}
```
